samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/ui/settings/SettingsViewModel.kt

```diff
@@ -56,23 +59,23 @@ class SettingsViewModel @Inject constructor(
 
     private fun validateCustomTrackUrl(trackUrl: String?): String {
         // Null check
-        if (trackUrl == null) {
+        if (trackUrl.isNullOrBlank()) {
             return ""
         }
-        // Protocol validation
-        if (!trackUrl.startsWith("http://") && !trackUrl.startsWith("https://")) {
-            return "URL must start with 'http://' or 'https://'"
-        }
-        // Host validation
-        if (Uri.parse(trackUrl)?.authority.isNullOrBlank()) {
-            return "Host must be defined"
-        }
-        // Ending character validation
-        if (!trackUrl.endsWith("/")) {
-            return "URL must end with '/'"
-        }
-        // Passed all checks, return empty string
-        return ""
+
+        return runCatching {
+            val uri = Uri.parse(trackUrl)
+            // Since SDK does not support custom schemes, we manually append http:// to the URL
+            // So the URL is considered invalid if it ends with a slash, contains a scheme, query or fragment
+            return@runCatching when {
+                uri.scheme != null -> "URL should not include 'http://' or 'https://'."
```

Comment:
I think it's good for the app to allow the flexibility of setting a scheme. Is there a reason why we are changing this behavior from before?

Reply:
The reason is the way analytics utilizing the values of [`apiHost`](https://github.com/segmentio/analytics-kotlin/blob/d2b3f183125efcd113de9fa0485795900162489b/core/src/main/java/com/segment/analytics/kotlin/core/HTTPClient.kt#L154) and [`cdnHost`](https://github.com/segmentio/analytics-kotlin/blob/d2b3f183125efcd113de9fa0485795900162489b/core/src/main/java/com/segment/analytics/kotlin/core/HTTPClient.kt#L143). To be more flexible here, we actually have to do a bit more parsing before passing the value matching what analytics need.

So I kept it simple to pass the exact value because

- This is sample app and not SDK API
- It is easier to write without `http` prefix than with it 😄